sdk/src/test/java/io/opentelemetry/sdk/trace/export/BatchSampledSpansProcessorTest.java

```diff
@@ -0,0 +1,349 @@
+    tracerSdk.addSpanProcessor(
+        BatchSampledSpansProcessor.newBuilder(waitingSpanExporter)
+            .setScheduleDelayMillis(MAX_SCHEDULE_DELAY_MILLIS)
+            .build());
+
+    createNotSampledEndedSpan(SPAN_NAME_1);
+    createNotSampledEndedSpan(SPAN_NAME_2);
+    ReadableSpan span2 = createSampledEndedSpan(SPAN_NAME_2);
+    // Spans are recorded and exported in the same order as they are ended, we test that a non
+    // sampled span is not exported by creating and ending a sampled span after a non sampled span
+    // and checking that the first exported span is the sampled span (the non sampled did not get
+    // exported).
+    List<Span> exported = waitingSpanExporter.waitForExport(1);
+    // Need to check this because otherwise the variable span1 is unused, other option is to not
+    // have a span1 variable.
+    assertThat(exported).containsExactly(span2.toSpanProto());
+  }
+
+  @Test(timeout = 10000L)
+  public void shutdownFlushes() {
+    // Set the export delay to zero, for no timeout, in order to confirm the #flush() below works
+    tracerSdk.addSpanProcessor(
+        BatchSampledSpansProcessor.newBuilder(waitingSpanExporter)
+            .setScheduleDelayMillis(0)
+            .build());
+
+    ReadableSpan span2 = createSampledEndedSpan(SPAN_NAME_2);
+
+    // Force a shutdown, without this, the #waitForExport() call below would block indefinitely.
+    tracerSdk.shutdown();
+
+    List<Span> exported = waitingSpanExporter.waitForExport(1);
+    assertThat(exported).containsExactly(span2.toSpanProto());
+  }
+
+  private static final class BlockingSpanExporter implements SpanExporter {
+    final Object monitor = new Object();
+
+    private enum State {
+      WAIT_TO_BLOCK,
+      BLOCKED,
+      UNBLOCKED
+    }
+
+    @GuardedBy("monitor")
+    State state = State.WAIT_TO_BLOCK;
+
+    @Override
+    public void export(List<Span> spanDataList) {
+      synchronized (monitor) {
+        while (state != State.UNBLOCKED) {
+          try {
+            state = State.BLOCKED;
+            // Some threads may wait for Blocked State.
+            monitor.notifyAll();
+            monitor.wait();
+          } catch (InterruptedException e) {
+            // Do nothing
+          }
+        }
+      }
+    }
+
+    private void waitUntilIsBlocked() {
+      synchronized (monitor) {
+        while (state != State.BLOCKED) {
+          try {
+            monitor.wait();
+          } catch (InterruptedException e) {
+            // Do nothing
+          }
+        }
+      }
+    }
+
+    @Override
+    public void shutdown() {
+      // Do nothing;
+    }
+
+    private void unblock() {
+      synchronized (monitor) {
+        state = State.UNBLOCKED;
+        monitor.notifyAll();
+      }
+    }
+  }
+
+  private static final class WaitingSpanExporter implements SpanExporter {
+    private final Object monitor = new Object();
+
+    @GuardedBy("monitor")
+    @SuppressWarnings("JdkObsolete")
```

Comment:
Does this apply here?

Reply:
Copy paste error. Was here because we had it as LinkedList.